### PR TITLE
Dev branch PR for code review to merge into main

### PR DESCRIPTION
**Pull request by theshivamverma**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+app.use("/product", productRouter);
+app.use("/cart", cartRouter);
+app.use("/user", userRouter);
+app.use("/wishlist", wishListRouter);
+app.use("/auth", authRouter);
+
+app.get("/", (req, res) => {
+  res.send("Hello world");
+});
+
```
Comment (h-harsh) [inline index.js:35]: You can put a route for error handleing

File: routes/cart.route.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    const cart = await Cart.find({}).populate({
+      path: "products",
+      populate: {
+        path: "product",
+        model: "Product",
+      },
+    });
+    res.json({ success: true, cart });
+  } catch (err) {
+    res.json({ success: false, message: "error retrieving carts data" });
```
Comment (h-harsh) [inline routes/cart.route.js]: Try giving status codes for error

File: routes/cart.route.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+  try {
+    const cart = await Cart.findById(id).populate({
+      path: "products",
+      populate: {
+        path: "product",
+        model: "Product",
+      },
+    });
+    if (!cart) {
+      res.status(400).jsn({ success: false, message: "error retrieving cart" });
```
Comment (h-harsh) [inline routes/cart.route.js]: correct spelling of JSON

File: routes/cart.route.js [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+             { "products._id": cartItem._id },
+             {
+               $set: {
+                 "products.$.quantity": 1,
+                 "products.$.visible": true
+               },
+             },
+             function (err, doc) {
+               if (err) {
+                 console.log(err);
```
Comment (h-harsh) [inline routes/cart.route.js]: Remove  console statements from code

File: routes/cart.route.js [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+  //   } catch (err) {
+  //     res
+  //       .status(400)
+  //       .json({
+  //         success: false,
+  //         message: "error updating cart",
+  //         errorMessage: err.message,
+  //       });
+  //   }
+  // });
```
Comment (h-harsh) [inline routes/cart.route.js]: remove unused code

File: routes/product.route.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  })
+  .post(async (req, res) => {
+    try {
+      const product = req.body;
+      const NewProduct = new Product(product);
+      const savedProduct = await NewProduct.save();
+      res.json({ success: true, product: savedProduct });
+    } catch (err) {
+      res
+        .status(500)
```
Comment (h-harsh) [inline routes/product.route.js]: check status code for this error

**Review by h-harsh — CHANGES_REQUESTED** [review]

File: index.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
+const express = require("express");
+const bodyParser = require("body-parser");
```
Comment (adarshwritescode) [inline index.js:2]: we can now use express's body parser

```
app.use(express.json())
```

File: routes/cart.route.js [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+    let { cart } = req;
+    res.json({ success: true, cart });
+  })
+  .post(async (req, res) => {
+    try {
+      const { prodId } = req.body;
+      let { cart } = req;
+      let flag = false
+      cart.products.map(cartItem => {
+        if(cartItem.product._id == prodId){
```
Comment (adarshwritescode) [inline routes/cart.route.js]: `===` convention

File: routes/product.route.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+  data.forEach(async (product) => {
+    try {
+      const NewProduct = await new Product(product);
+      const savedProduct = await NewProduct.save();
+      console.log({ savedProduct });
+    } catch (err) {
+      console.log({ err });
+    }
+  });
+}
```
Comment (adarshwritescode) [inline routes/product.route.js]: you can comment it out if this is of no use

File: routes/user.route.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    try {
+      const userData = await User.find({}).select('username email -_id')
+      res.json({ success: true, userData });
+    } catch (err) {
+      res.json({ success: false, errorMessage: err.message });
+    }
+  })
+  .post(async (req, res) => {
+    try {
+      const user = req.body;
```
Comment (adarshwritescode) [inline routes/user.route.js]: sanitizing body before storing would be great :)

you can trim whitespaces around and convert it to lower case

**Pull request merged**